test/extended/prometheus/prometheus.go

```diff
@@ -49,6 +53,120 @@ type TelemeterClientConfig struct {
 	Enabled *bool `json:"enabled"`
 }
 
+var componentNamespace = os.Getenv("COMPONENT_NAMESPACE")
+
+var _ = g.Describe("[sig-instrumentation][Late] OpenShift service monitors [apigroup:image.openshift.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc                         = exutil.NewCLIWithPodSecurityLevel("prometheus", admissionapi.LevelBaseline)
+		prometheusURL, bearerToken string
+
+		// TODO: remove the namespace when the bug is fixed.
+		namespacesToSkip = sets.New[string]("openshift-monitoring", // https://issues.redhat.com/browse/MON-4304
+			"openshift-marketplace",                // https://issues.redhat.com/browse/OCPBUGS-59763
+			"openshift-image-registry",             // https://issues.redhat.com/browse/OCPBUGS-59767
+			"openshift-operator-lifecycle-manager", // https://issues.redhat.com/browse/OCPBUGS-59768
+			"openshift-cluster-samples-operator",   // https://issues.redhat.com/browse/OCPBUGS-59769
+			"openshift-cluster-version",            // https://issues.redhat.com/browse/OCPBUGS-57585
+		)
+	)
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		var err error
+		prometheusURL, err = helper.PrometheusRouteURL(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Get public url of prometheus")
+		bearerToken, err = helper.RequestPrometheusServiceAccountAPIToken(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Request prometheus service account API token")
+
+		if namespacesToSkip.Has(componentNamespace) {
+			e2e.Logf("The namespace %s is not skipped because $COMPONENT_NAMESPACE is set to it", componentNamespace)
+			namespacesToSkip.Delete(componentNamespace)
+		}
+	})
+
+	g.It("should not be accessible without authorization [Serial]", func() {
+		var errs []error
+
+		g.By("verifying all targets returns 401 or 403 or 500 without authorization")
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), "execpod-targets-authorization")
+		defer func() {
+			err := oc.AdminKubeClient().CoreV1().Pods(execPod.Namespace).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("Delete pod %s/%s", execPod.Namespace, execPod.Name))
+		}()
+
+		contents, err := helper.GetURLWithToken(helper.MustJoinUrlPath(prometheusURL, "api/v1/targets"), bearerToken)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		targets := &prometheusTargets{}
```

Comment:
we can add a sanity test, that there are at least 5 targets for example.

Reply:
Done.